src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -209,22 +217,25 @@ def get_context(group: Group) -> str:
 
     state = None
     event_count = None
-    if "State" in context:
-        state = SUPPORTED_CONTEXT_DATA["State"](group)
-    if "Events" in context:
-        event_count = SUPPORTED_CONTEXT_DATA["Events"](group)
+    if NotificationContextField.STATE in context:
+        state = SUPPORTED_CONTEXT_DATA[NotificationContextField.STATE](group)
+    if NotificationContextField.EVENTS in context:
+        event_count = SUPPORTED_CONTEXT_DATA[NotificationContextField.EVENTS](group)
 
     if (state and state == "New") or (event_count and int(event_count) <= 1):
-        if "Events" in context:
-            context.remove("Events")
+        if NotificationContextField.EVENTS in context:
+            context.remove(NotificationContextField.EVENTS)
 
         # avoid hitting Snuba for user count if we don't need it
-        if "Users Affected" in context:
-            context.remove("Users Affected")
+        if NotificationContextField.USERS_AFFECTED in context:
+            context.remove(NotificationContextField.USERS_AFFECTED)
 
     for c in context:
         if c in SUPPORTED_CONTEXT_DATA:
-            v = SUPPORTED_CONTEXT_DATA[c](group)
+            if c == NotificationContextField.USERS_AFFECTED:
+                v = SUPPORTED_CONTEXT_DATA[c](group, rules)
```

Comment:
this is also messy and i'm not sure if there's a better way to do this without making every other function in `SUPPORTED_CONTEXT_DATA` take in an unused `rules` argument

Reply:
I don't have enough context on this to have a strong opinion, but you could make the argument that updating each definition to include rules is preferable. Mainly because it keeps the definitions consistent while providing access to rules for future cases similar to this one.